tests/test_CSRGraphFromCoordinates.cpp

```diff
@@ -0,0 +1,375 @@
+/*
+  Copyright 2023 Equinor
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#define BOOST_TEST_MODULE CSR_Graph_From_Coordinates
+
+#include <boost/test/unit_test.hpp>
+
+#include <opm/common/utility/CSRGraphFromCoordinates.hpp>
+
+#include <cstddef>
+
+BOOST_AUTO_TEST_SUITE(Untracked)
```

Comment:
One final remark: I think a test exercising the merge() method should be added (for both tracked and untracked) where some edges are added to an existing graph (both duplicates of existing ones, and entirely new ones).

Reply:
> One final remark: I think a test exercising the `merge()` method should be added (for both tracked and untracked) where some edges are added to an existing graph (both duplicates of existing ones, and entirely new ones).

I've expanded the test suite to cover these cases, I think.